sdk/schemaregistry/schema-registry/src/operations.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RestError } from "@azure/core-rest-pipeline";
+import { isUnexpected } from "./generated/isUnexpected";
+import {
+  GetSchemaOptions,
+  GetSchemaPropertiesOptions,
+  RegisterSchemaOptions,
+  Schema,
+  SchemaDescription,
+  SchemaProperties,
+} from "./models";
+import { buildContentType, convertSchemaIdResponse, convertSchemaResponse } from "./conversions";
+import { SchemaRegistryClient } from "./generated/clientDefinitions";
+
+export async function registerSchema(
+  context: SchemaRegistryClient,
+  schema: SchemaDescription,
+  options: RegisterSchemaOptions = {},
+): Promise<SchemaProperties> {
+  const { groupName, name: schemaName, definition: schemaContent, format } = schema;
+  const response = await context
+    .path("/$schemaGroups/{groupName}/schemas/{schemaName}", groupName, schemaName)
+    .put({
+      contentType: buildContentType(format) as any,
```

Comment:
could we make the return type of buildContentType more precise to get rid of the cast to any?